exist-core/src/main/java/org/exist/xquery/functions/util/DocumentNameOrId.java

```diff
@@ -106,98 +104,103 @@ public DocumentNameOrId(final XQueryContext context, final FunctionSignature sig
-            absoluteId = absoluteId.or(BigInteger.valueOf(doc.getResourceType() & 1));
-            result = new IntegerValue(absoluteId, Type.INTEGER);
-        } else {
-            result = Sequence.EMPTY_SEQUENCE;
+    private @Nullable <T> T getFromDocument(final XmldbURI docUri, final Function<DocumentImpl, T> docFn) throws PermissionDeniedException {
+        try (final LockedDocument lockedDoc = context.getBroker().getXMLResource(docUri, LockMode.READ_LOCK)) {
+            if (lockedDoc == null) {
+                return null;
+            } else {
+                return docFn.apply(lockedDoc.getDocument());
+            }
         }
+    }
 
-        return result;
+    private static BigInteger getAbsoluteResourceId(final DocumentImpl doc) {
+        return encodeAbsoluteResourceId(doc.getCollection().getId(), doc.getDocId(), doc.getResourceType());
     }
 
-    private Sequence resourceById(final Sequence args[]) throws IOException, PermissionDeniedException, XPathException, LockException {
-        final String fnName = getSignature().getName().getLocalPart();
-        if (fnName.equals(FSN_GET_RESOURCE_BY_ABSOLUTE_ID)) {
-            final IntegerValue absoluteIdParam = (IntegerValue) args[0].itemAt(0);
-            BigInteger absoluteId = absoluteIdParam.toJavaObject(BigInteger.class);
-
-            final byte resourceType = absoluteId.testBit(0) ? DocumentImpl.BINARY_FILE : DocumentImpl.XML_FILE;
-            absoluteId = absoluteId.shiftRight(1);
-            final int documentId = absoluteId.and(BigInteger.valueOf(0xFFFFFFFF)).intValue();
-            absoluteId = absoluteId.shiftRight(32);
-            final int collectionId = absoluteId.and(BigInteger.valueOf(0xFFFFFFFF)).intValue();
-
-            final DocumentImpl doc = context.getBroker().getResourceById(collectionId, resourceType, documentId);
-            if (doc != null) {
-                try (final ManagedDocumentLock docLock = context.getBroker().getBrokerPool().getLockManager().acquireDocumentReadLock(doc.getURI())) {
-                    if (doc instanceof BinaryDocument) {
-                        final BinaryDocument bin = (BinaryDocument) doc;
-                        final InputStream is = context.getBroker().getBinaryResource(bin);
-                        final Base64BinaryDocument b64doc = Base64BinaryDocument.getInstance(context, is);
-                        return b64doc;
-                    } else {
-                        return new NodeProxy(doc);
-                    }
+    private Sequence getResourceByAbsoluteId(final IntegerValue ivAbsoluteId)
+            throws XPathException, PermissionDeniedException, LockException, IOException {
+        final BigInteger absoluteId = ivAbsoluteId.toJavaObject(BigInteger.class);
+        final Tuple3<Integer, Integer, Byte> decoded = decodeAbsoluteResourceId(absoluteId);
+
+        final DocumentImpl doc = context.getBroker().getResourceById(decoded._1, decoded._3, decoded._2);
+        if (doc != null) {
```

Comment:
Could this be rewritten as:
```java
        if (doc == null) { return Sequence.EMPTY_SEQUENCE; }

        try (final ManagedDocumentLock docLock = context.getBroker().getBrokerPool().getLockManager().acquireDocumentReadLock(doc.getURI())) {
            if (doc instanceof BinaryDocument) {
                final BinaryDocument bin = (BinaryDocument) doc;
                final InputStream is = context.getBroker().getBinaryResource(bin);
                return Base64BinaryDocument.getInstance(context, is);
            }
            return new NodeProxy(doc);
        }

        return Sequence.EMPTY_SEQUENCE;
```

Reply:
Yes it could, but that is not clearer to me.

We seem to spend a lot of time lately discussing code-style. Perhaps if this is a concern we should just adopted a standard like: https://google.github.io/styleguide/javaguide.html. Otherwise we are just arguing subjectively about personal taste.

Reply:
Good topic for Monday 👍